src/RedisRateLimiting/TokenBucket/RedisTokenBucketRateLimiter.cs

```diff
@@ -14,7 +15,12 @@ public class RedisTokenBucketRateLimiter<TKey> : RateLimiter
 
         private readonly TokenBucketLease FailedLease = new(isAcquired: false, null);
 
-        public override TimeSpan? IdleDuration => TimeSpan.Zero;
+        private int _activeRequestsCount;
+        private long _lastActiveTimestamp = Stopwatch.GetTimestamp();
+
+        public override TimeSpan? IdleDuration => Interlocked.CompareExchange(ref _activeRequestsCount, 0, 0) > 0
```

Comment:
Quick q: why do we need to track the number of active requests? Can't we just rely on the `_idleSince` timestamp?

Reply:
This is to ensure that during `AcquireAsyncCoreInternal` execution we will never report we are idle when we don't.

Theoretically if we have infrequent usage of that rate limiter and for some reason Redis connection is slow and request takes 10+ seconds, than we will report 10+ seconds to .Net and then there is a chance .Net will decide that our rate limiter instance is idle and needs to be disposed, even though we are waiting for AcquireAsyncCoreInternal to complete

Probably, that's an overkill and in reality Redis shouldn't be that slow, but I didn't want to make such assumptions for common library.